### ISSUE TITLE
NETSDK1056: Project is targeting runtime 'win-x64' but did not resolve any runtime-specific packages. This runtime may not be supported by the target framework.	ILTools 

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#37582]
**Version used**: VisualStudio.16.Preview/16.3.0-pre.1.0+29123.89
C# Tools 3.3.0-beta1-19360-03+7d47461d2ddd32defbc51c2cf0e75a3fef16c3d7

I have been getting this error occasionally on incremental builds of Roslyn.sln. It doesn't seem to stop me from running my tests but it is a bit confusing.

`NETSDK1056: Project is targeting runtime 'win-x64' but did not resolve any runtime-specific packages. This runtime may not be supported by the target framework.	ILTools	F:\src\roslyn\src\Tools\ILTools\ILTools.proj	1`

**Comment by RikkiGibson** [comment]
This error also seems to have a habit of breeding, after working for a while I saw 11 dupes of the same one in the VS error list

**Comment by sharwell** [comment]
I sometimes get this, but then it disappears after the IDE restores packages. I was guessing it's a difference in behavior between command line and IDE restore.

**Comment by chborl** [comment]
I'm getting this in the IDE today too.

**Comment by dibarbet** [comment]
+1 from me, been seeing this intermittently throughout the week.

**Comment by tmat** [comment]
@nguerrera @jaredpar Any ideas?

**Comment by jaredpar** [comment]
This is the PR that will fix this, and many other MSBuild warnings, in our code base

> https://github.com/dotnet/roslyn/pull/37666

It's been active for five days without review. Maybe we should review it :smile: 

Side note: this warning, and the others, crept into our code base due to the massive amount of IDE style warnings that are present. These style warnings prevent us from enabling MSBuild's warning as error feature (for non-compiler warnings). Further there are so many individual developers don't notice when new warnings appear. Had MSBuild's warn as error been enabled this would've never gotten into our code base.

We need to find a way to structure the IDE style warnings such that they don't compromise the correctness of our code base.

**Comment by jaredpar** [comment]
Note: I don't 100% know if this fixes this warning. I was actually focused on our MSB3270 warnings. Again though so many IDE style warnings it was hard to see if any relevant warnings remained after. If this warning is still present I can look into it.

**Comment by chborl** [comment]
@jaredpar , I merged in these changes and I'm still seeing the error message.

**Comment by jaredpar** [comment]
@chborl how do you see them? I tried a couple of command line builds and saw nothing with these changes.

**Comment by chborl** [comment]
They only happen in VS builds.  Both the public preview release and int preview.

**Comment by nguerrera** [comment]
Looking at ILTools.proj and the code that resolves it, it's behaving as though restore did not see the RuntimeIdentifier in the project.

If you can use https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProjectSystemTools to gather a binlog of the build that produced this warning, and also grab project.assets.json for ILTools.proj, that would help to confirm that theory and also to try to find hints as to why.

**Comment by nguerrera** [comment]
cc @dsplaisted Maybe this is related tot he work to allow design time builds to pass while assets file is out of date? Does this warning get falsely reported while restore is in progress?

**Comment by chborl** [comment]
@nguerrera , Is this what you need for binlog?
[BuildLogs.zip](https://github.com/dotnet/roslyn/files/3475174/BuildLogs.zip)

project.assets.json renamed with txt extension:
[project.assets.txt](https://github.com/dotnet/roslyn/files/3475197/project.assets.txt)

**Comment by nguerrera** [comment]
@davkean @nkolev92 somehow we’ve been given an assets file for a 2.1 project with no package assets at all. Seems like an issue with nomination or restore.

**Comment by chborl** [comment]
Actually, this is the correct project.assets.json
[project.assets.txt](https://github.com/dotnet/roslyn/files/3475250/project.assets.txt)

**Comment by nkolev92** [comment]
I don't see the binlog for the ILTools proj. Am I missing something? 

If this repros somewhat often, you can try setting the following en var PROJECTSYSTEM_PROJECTOUTPUTPANEENABLED=1 and re-launching VS. 

It will dump the nomination data in something similar to an output window. Then we might be able to narrow down where things went wrong.

**Comment by davkean** [comment]
This project is a custom project that is pretending to be a C# project. The design-time build is failing which results in zero package references:
[ILTools.zip](https://github.com/dotnet/roslyn/files/3475329/ILTools.zip)

I cannot tell why the project is failing, maybe missing core targets?

**Comment by davkean** [comment]
Wrong log, here is the correct one:

[ILTools_Debug_AnyCPU_DesignTimeBuild_2019-08-07T14_58_00.6011148+10_00.zip](https://github.com/dotnet/roslyn/files/3475330/ILTools_Debug_AnyCPU_DesignTimeBuild_2019-08-07T14_58_00.6011148%2B10_00.zip)

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\master\MSBuild\Microsoft\VisualStudio\Managed\Microsoft.Managed.DesignTime.targets(423,54): error MSB4057: The target "CompileDesignTime" does not exist in the project. [C:\roslyn\src\Tools\ILTools\ILTools.proj]
```

**Comment by davkean** [comment]
This project isn't importing the C# or VB design-time targets, and looks like there's a dangling reference from Managed.DesignTime.targets to that target.

**Comment by jaredpar** [comment]
> This project isn't importing the C# or VB design-time targets 

Yes because that causes other problems like forcing outputs. I can't remember all the details but eventually we removed them to work past one of the thousand paper cuts we got making this package function.

The history here is this package is basically impossible to get correct. This package must be paired with an exact version of the runtime in order to function. Further runtime package will never be the same runtime that we install as a part of our repository setup.

The last time we had to upgrade this package I think it took about one week, some deep discussion with @ericstj, ample profanity and eventually we found this was the only workable solution.

This setup has been around for some time though. Why did this warning just start appearing?

Side note: CoreCLR recently make ilasm / ildasm independent of the runtime. So there is hope this will get simpler for us in the near future. But probably a few weeks out. https://github.com/dotnet/coreclr/pull/25930
